utils/src/main/kotlin/Extensions.kt

```diff
@@ -60,7 +60,20 @@ fun File.expandTilde(): File = File(path.expandTilde()).absoluteFile
  * Return the hexadecimal digest of the given hash [algorithm] for this [File].
  */
 fun File.hash(algorithm: String = "SHA-1"): String =
-    MessageDigest.getInstance(algorithm).digest(readBytes()).toHexString()
+    inputStream().use { inputStream ->
+        // 4MB has been chosen rather arbitrary hoping that it provides a good enough performance while not consuming
+        // too a lot of memory, also considering that this function could potentially be run on multiple thread in
+        // parallel.
```

Comment:
"while not consuming too a lot of memory" sounds weird, maybe "while not consuming a lot of memory at the same time".
"multiple thread**s**"

Reply:
I just copied this overly blindly, but I've fixed it now away.